### ISSUE TITLE
Add a option to upload a specific folder to sharepoint instead of an archive of the repository

### ISSUE THREAD

**Issue by harsohailB** [obrassard/action-sharepoint-publish#6]
I am trying to build a react app and deploy the build folder to a SharePoint document library. As this GitHub action creates a zip, is there a way to pass environment variables to change the filename to a folder instead of a zip that consists of the build files?

**Comment by obrassard** [comment]
Hi @harsohailB, I hope your doing good ! 

In fact, for the moment, the action does not allow to upload anything else than the zip file containing the repo. The action was created for a very specific need at the beginning. However, I think it could indeed be a great improvement to be able to upload a complete folder on sharepoint without compression.

I use the [`spsave` ](https://www.npmjs.com/package/spsave) module in order to upload files to sharepoint. To my knowledge it seems that this module doesn't directly support uploading a folder, only files.. So, I think the solution would be to add a new optional parameter to the action which specify a folder to upload (for instance your build folder), and to update `index.js` to loop over the given folder content (and upload individual files).. 

I don't know when I'll have time to work on these changes though, but if you want to do a PR you're welcome ! Let me know !😉

**Comment by obrassard** [comment]
FYI, I asked a question on spsave repo to check if they already support folder upload : https://github.com/s-KaiNet/spsave/issues/63

**Comment by harsohailB** [comment]
@obrassard Thank you for the quick response! The `sp save` module information was very useful and I was able to fork the repo to make the changes fit for my project. You have done an amazing job with this project and have saved me so much time with this action!

Here is the repo I created with changes to push a specific build folder to SharePoint:
https://github.com/harsohailB/action-sharepoint-publish

**Comment by obrassard** [comment]
Hey @harsohailB i'm glad to hear that you were able to make it work!! 

I can see that we don't need to iterate through every files, thanks to the `glob` option, is it right ?

**Comment by harsohailB** [comment]
@obrassard Yes, that's correct. The glob option chooses all files from within a folder. Using wildcards in the glob like `build/**/*.*` allows us to preserve folder structure within the build folder. This way all imports work as they are intended to for the react app.

**Comment by obrassard** [comment]
Nice thanks !! I will update my own action in order to add more flexibility zip of repo / target directory !

**Comment by av1v3k** [comment]
Hi @obrassard,

How do I generate clientID and client Secret ? Do I need to have Azure account for this ?

 I was going through this article. 

https://learn.microsoft.com/en-us/sharepoint/dev/solution-guidance/security-apponly


 Please help.

**Comment by obrassard** [comment]
> Hi @obrassard,
> 
> 
> 
> How do I generate clientID and client Secret ? Do I need to have Azure account for this ?
> 
> 
> 
>  I was going through this article. 
> 
> 
> 
> https://learn.microsoft.com/en-us/sharepoint/dev/solution-guidance/security-apponly
> 
> 
> 
> 
> 
>  Please help. 

Hello @purplebearddan do you know what's the recommended way to generate a client ID / secret for SharePoint? Thanks in advance

**Comment by purplebearddan** [comment]
This link is what our IT admin used to gen the deets required

https://learn.microsoft.com/en-us/sharepoint/dev/solution-guidance/security-apponly-azureacs

@flavioPBadmin do you have any input for this?

**Comment by bobkaru** [comment]
> @obrassard Thank you for the quick response! The `sp save` module information was very useful and I was able to fork the repo to make the changes fit for my project. You have done an amazing job with this project and have saved me so much time with this action!
> 
> Here is the repo I created with changes to push a specific build folder to SharePoint: https://github.com/harsohailB/action-sharepoint-publish

Hello harsohailB, I'm new to GitHub and I'm trying to deploy the only modify/ added files from the GitHub repo to SharePoint folders. I followed the instruction and used the same .yml script, however, I'm getting the error @ Build page name "no such file or directory, open '/home/runner/work/.../package.json" with error code 254 any clue?
